ape_ledger/accounts.py

```diff
@@ -84,12 +88,55 @@ def sign_message(self, msg: SignableMessage) -> Optional[MessageSignature]:
             signed_msg = self._client.sign_typed_data(msg.header, msg.body)
         else:
             raise LedgerSigningError(
-                f"Unsupported message-signing specification, (version={version!r})"
+                f"Unsupported message-signing specification, (version={version!r})."
             )
 
-        return MessageSignature(*signed_msg)  # type: ignore
+        v, r, s = signed_msg
+
+        if self.provider:
```

Comment:
I am not happy with this part but I don't know how else to do it.
Basically, it seems like we need to do some modulo math on `v` for some chain IDs.

https://github.com/LedgerHQ/app-ethereum/blob/develop/examples/signMessageEIP711v0.py#L75-L81
https://github.com/LedgerHQ/app-ethereum/blob/develop/examples/signMessage.py#L75-L81

So we need to know the chain ID at this point, but the user is not connected to a provider, who knows what they heck they are signing....

At least in `sign_transaction`, chain ID is set on the transaction. Here, we can't assume the chain ID is in the message anywhere so we have to check the active provider.

Reply:
This logic seems pretty appropiate to me actually